core/state_transition.go

```diff
@@ -566,10 +576,11 @@ func (st *stateTransition) buyGas() error {
 			// when both SGT and native balance are used, we record both amounts for refund.
 			// the priority for refund is: first native, then SGT
 			usedNativeBalance := new(uint256.Int).Sub(mgvalU256, soulBalance)
-			if usedNativeBalance.Sign() > 0 {
-				st.state.SubBalance(st.msg.From, usedNativeBalance, tracing.BalanceDecreaseGasBuy)
-				st.usedNativeBalance = usedNativeBalance
+			if usedNativeBalance.Sign() <= 0 {
```

Comment:
Looks like this will never happen, `usedNativeBalance` is always positive.

Reply:
done